pkg/controllers/queue/queue_controller.go

```diff
@@ -137,40 +139,50 @@ func (c *Controller) processNextWorkItem() bool {
 	return true
 }
 
-func (c *Controller) syncQueue(key string) error {
-	glog.V(4).Infof("Begin sync queue %s", key)
-
-	var pending, running, unknown, inqueue int32
+func (c *Controller) getPodGroups(key string) ([]string, error) {
 	c.pgMutex.RLock()
+	defer c.pgMutex.RUnlock()
+
 	if c.podGroups[key] == nil {
-		c.pgMutex.RUnlock()
-		glog.V(2).Infof("queue %s has not been seen or deleted", key)
-		return nil
+		return nil, fmt.Errorf("queue %s has not been seen or deleted", key)
```

Comment:
It is dangerous, think a queue is deleted before a pg. It will cause endless retry.

Reply:
let's handle this in different PR, refer to the TODO in following codes :)